test/Microsoft.Extensions.Logging.Test/ConsoleLoggerTest.cs

```diff
@@ -54,16 +54,27 @@ public ConsoleLoggerTest()
         }
 
         [Fact]
-        public void ThrowsException_WhenNoMessageAndExceptionAreProvided()
+        public void LogsWhenMessageIsNotProvided()
         {
             // Arrange
             var t = SetUp(null);
             var logger = (ILogger)t.Item1;
             var sink = t.Item2;
+            var exception = new InvalidOperationException("Invalid value");
+            var expectedMessage = _paddingString + "[null]" + Environment.NewLine;
+            var expectedExceptionMessage =
+                exception.ToString() + Environment.NewLine;
 
-            // Act & Assert
-            var exception = Assert.Throws<ArgumentNullException>(() => logger.LogCritical(eventId: 0, exception: null, message: null));
-            exception = Assert.Throws<ArgumentNullException>(() => logger.LogCritical(eventId: 0, message: null));
+            // Act
+            logger.LogCritical(eventId: 0, exception: null, message: null);
+            logger.LogCritical(eventId: 0, message: null);
+            logger.LogCritical(eventId: 0, message: null, exception: exception);
+
+            // Assert
+            Assert.Equal(expectedMessage, sink.Writes[2].Message);
+            Assert.Equal(expectedMessage, sink.Writes[5].Message);
+            Assert.Equal(expectedMessage, sink.Writes[8].Message);
+            Assert.Equal(expectedExceptionMessage, sink.Writes[9].Message);
```

Comment:
Magic numbers 😁  Isn't there a less-potentially-flaky way to capture those messages?